carbon-design-system/carbon-addons-iot-react
# feat(timepicker): design tweaks

Closes #3566 
Closes #3515 
Closes #3575 

**Summary**

- Add support for 24 hours format
- Fix issues from design audit

**Change List (commits, features, bugs, etc)**

- Enable 24 hours format selection (with validation)
- Design tweaks from audit (see full list in Acceptance Test)
- Update tests and storyshots

**Acceptance Test (how to verify the PR)**

- Support for 24 hours format
  - Go to [this story](https://deploy-preview-3584--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-timepickerdropdown--default)
  - Enable knob for `is24hours` format
  - Verify that time picker has changed its format to 24 hours
- Label width issue for time range selector
  - Go to [this story](https://deploy-preview-3584--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-timepickerdropdown--default)
  - Select type `range`
  - Hide secondary label
  - Type some long string in `i18n` knob in `labelText` field
  - Verify that label spans across 2 time pickers
- Single focus state on time input (design audit 1)
  - Go to [this story](https://deploy-preview-3584--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-timepickerdropdown--default)
  - Set focus on time input
  - Verify that dropdown has appeared
  - Verify that current time has been populated
  - Verify that focus preserved in the input field
  - Press `ArrowDown` in order to move focus in dropdown
  - Use arrow keys to select some date
  - Press `Enter` to return focus back to input field
 - Dropdown arrows color on hover (design audit 2)
   - Go to [this story](https://deploy-preview-3584--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-timepickerdropdown--time-picker-spinner-story)
   - Hover on some arrow icon button
   - Verify that color of icon has not changed
  - Replace border with outline (design audit 3)
    - Go to [this story](https://deploy-preview-3584--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-timepickerdropdown--default)
    - Focus on input field
    - Move focus to dropdown (by clicking on some number)
    - Verify that `1px` outline has appeared on input and text position has not changed
  - Placeholder for meridiem (design audit 4)
    - Go to [this story](https://deploy-preview-3584--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-timepickerdropdown--default)
    - Verify that placeholder for input includes `hh:mm XM`
  - Separate warning and error state for range input (design audit 5)
    - Go to [this story](https://deploy-preview-3584--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-timepickerdropdown--default)
    - Select type `range`
    - Verify that 2 helper texts has appeared
    - Enable `Second input invalid` knob
    - Verify that only 2nd input has changed
    - Enable `Show second input warning state` knob
    - Verify that only 2nd input has changed
 - Add scrolling for meridiem (design audit 6)
   - Go to [this story](https://deploy-preview-3584--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-timepickerdropdown--time-picker-spinner-story)
   - Click on the input field
   - Select different meridiem
   - Verify that its spinner has scrolled
 - Selected time background color on hover
   - Go to [this story](https://deploy-preview-3584--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-timepickerdropdown--time-picker-spinner-story)
   - Hover on the selected time
   - Verify that background color changed to `$hover-primary` color

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here



**Things to look for during review**
